CMakeLists.txt

```diff
@@ -371,12 +363,12 @@ foreach (family ${ARCH})
         if (USE_RUST)
             target_sources(${target} PRIVATE ${RUST_FILES})
```

Comment:
looks like this is still using `${RUST_FILES}` which was removed, so I think this would fail with USE_RUST. 

As no-one is actually using the Rust bindings for anything (and if they wanted to, there are probably better ways of doing this, in retrospect), we could also remove them if you think the build system support to keep them is disproportionate.

Reply:
I actually tested `-DUSE_RUST=ON` and it links and runs fine (including with the weird Rust-Python bridge enabled). I haven't tested calling them at runtime since I did get undefined symbol errors before I added the Rust support and that seemed enough.

It didn't cause me any significant effort to keep the support in; I suspect having them around probably doesn't make sense but this isn't a decision that I feel like should be taken based on build system complexity. Rather, should we remove them because they're not even tested on CI?

I can do the removal if you'd like me to.

Reply:
Thanks for confirming that it still builds fine, that's fine for this PR - I will probably think about removal (or adding a quick CI smoketest) once the build system refactoring is done.